### PR TITLE
Introduce APIService resource to Rancher

### PR DESCRIPTION
**Pull request by orangedeng**
**Problem:**
The custom metrics server associated with `pod/object/external`
types is not available by default, we need a way to tell UI we don't
support those kind of metrics for now.

**Solution:**
Introduce `apiservice.apiregistration.k8s.io` resource into Rancher.
This API can tell which api service is current available.

Related issue:
https://github.com/rancher/rancher/issues/20693

### CODE REVIEW

File: apis/cluster.cattle.io/v3/schema/schema.go [hunk 1]
```diff
@@ -130,3 +132,11 @@ func tokens(schemas *types.Schemas) *types.Schemas {
 			schema.ResourceMethods = []string{}
 		})
 }
+
+func apiservice(Schemas *types.Schemas) *types.Schemas {
```
Comment (gitlawr) [inline apis/cluster.cattle.io/v3/schema/schema.go]: Use camel case and plural for the function name

**Comment by gitlawr** [comment]
Regarding the origin issue, I think we don't have to add APIService to Rancher API. UI can know that by talking to kubernetes API via proxy: `<server-url>/k8s/clusters/<c-id>/apis`. It should be accessible by any users that can get the cluster object, including project-members.

**Comment by cjellick** [comment]
@gitlawr how would you then subscribe to the changes to the resource so that the UI would dynamically update whenan APIService is created/updated/deleted?

**Comment by gitlawr** [comment]
It can't do that if UI talks to the `/apis` endpoint. From this point, this PR is a proper solution.

File: vendor.conf [hunk 2]
```diff
@@ -4,3 +4,4 @@ github.com/rancher/types
 github.com/pkg/errors v0.8.0
 github.com/rancher/norman  f5744043a6fb81330ee78e4f7a0f04d0ef65c9f1  transitive=true 
 github.com/coreos/prometheus-operator         v0.25.0 
+k8s.io/kube-aggregator     kubernetes-1.12.5            https://github.com/kubernetes/kube-aggregator
```
[the remark this answers is not in the document]
Comment (orangedeng) [reply vendor.conf]: That’s because we use the same k8s version in Norman. At least the same breach name as the Norman is using for k8s.io.

**Comment by orangedeng** [comment]
Updated

**Comment by gitlawr** [comment]
LGTM

**Comment by orangedeng** [comment]
Rebased.

**Comment by cjellick** [comment]
@ibuildthecloud do you see any issues with this PR?

**Comment by ibuildthecloud** [comment]
LGTM

**Pull request merged**